### PR TITLE
Extended double-click options

### PR DESCRIPTION
**Pull request by Netroxen**
Extends the double-click option with the following options:
- None (single-click default)
- file (file-only double-click mode)
- all (file and folders double-click mode)

Changes:
- Field definition: `bool` -> `String`

- [x] Added an entry to `CHANGELOG.md` if this change could be valuable to users

### CODE REVIEW

File: lapce-ui/src/explorer.rs [hunk 1]
```diff
@@ -687,52 +689,72 @@ impl Widget<LapceTabData> for FileExplorerFileList {
...
+                        double_click_mode = 1;
+                    }
+                    "all" => {
+                        double_click_mode = 2;
+                    }
+                    _ => {
+                        double_click_mode = 0;
+                    }
+                }
+
```
Comment (panekj) [inline lapce-ui/src/explorer.rs:703]: this should be enum type rather than int

Comment (Netroxen) [reply lapce-ui/src/explorer.rs]: Please see latest commit - is this what you mean?

Comment (hbina) [reply lapce-ui/src/explorer.rs]: No, `EditorConfig` will be directly deserialized from the file so you can instead automatically generate the conversion using

```rust
#[derive(Serialize, Deserialize)]
enum ClickModes {
    #[serde(rename = "single")]
    SingleClick,
    #[serde(rename = "file")]
    DoubleClickFile,
    #[serde(rename = "all")]
    DoubleClickAll,
}
```

and put the enum directly,

```rust
#[field_names(
        desc = "Use a double click to interact with the file explorer.\nOptions: file, all or Nothing (Default)."
    )]
    pub double_click: ClickModes,
```

and set the default to "single".

This removes the "magic constants" from plaguing the codebase and keep
There doesn't seem to be any error handling yet on the  `ReloadConfig` command (it simply fallback to the default config on failure).

Comment (Netroxen) [reply lapce-ui/src/explorer.rs]: Ahhh thanks for the information, as you can probably tell, I'm relatively new to Rust - so this really helps 💪

Ideally a drop-down menu would be even nicer, but unfortunately that was a little too complicated for me at this stage.

Comment (hbina) [reply lapce-ui/src/explorer.rs]: Yea creating the dropdown menu is quite tricky. Feel free to ping me if you have any more questions.

Comment (Netroxen) [reply lapce-ui/src/explorer.rs]: Alright, I have changed it as requested - please take a look.

**Comment by Netroxen** [comment]
@panekj can be re-reviewed unless I missed something?

**Pull request merged**